compiler/jsgen.nim

```diff
@@ -2043,7 +2043,11 @@ proc genMagic(p: PProc, n: PNode, r: var TCompRes) =
   of mNew, mNewFinalize: genNew(p, n)
   of mChr: gen(p, n[1], r)
```

Comment:
optional (or future work):
add performance test in the recent `tests/benchmarks` directory (see its readme) to see if this does result in performance improvement (in at least some test cases)

Reply:
But if `tests/benchmarks` is not tested, that's not easy to maintain;

Reply:
we can discuss this further in https://github.com/timotheecour/Nim/issues/425; short answer:
* for now, you can at least have `action: compile` (as in tests/benchmarks/ttls.nim)
* in future, we can also test in CI by passing `--define:expectedDuration:numSeconds` flag in CI so the test runs "fast enough", which still allows running locally with more accurate (but longer) timing, and generate whatever statistics/plots across commit history to find regressions

Reply:
I agree.